src/test/java/com/datatorrent/netlet/util/CircularBufferTest.java

```diff
@@ -52,6 +52,57 @@ public void tearDown()
   }
 
   /**
+   * Test larger than allowed size of buffer; we expect an exception
+   */
+  @Test
+  public void testLargeSize()
+  {
+    try {
+      final int size = 1 + CircularBuffer.MAX_SIZE;
+      CircularBuffer<Integer> instance = new CircularBuffer<Integer>(size);
+      assert false;
+    } catch (RuntimeException e) {    // expected
+      System.out.format("Got expected exception: %s%n", e);
+    }
+  }
+
+  // helper routine to test that argument works and yields the same size
+  // as the old method
+  //
+  private void testSize(final int size)
+  {
+    CircularBuffer<Integer> instance = new CircularBuffer<Integer>(size);
+
+    // get size the old way and check that it matches
+    int i = 1;
+    while (i < size) {
+      i <<= 1;
+    }
+    assert i == instance.capacity();
+
+    // check that the instance is actually functional
+    instance.add(99);
+    final int v = instance.remove();
```

Comment:
use poll() and peek(). Check that the instance is empty after the last expected element is deleted from the CircularBuffer.